### PR TITLE
API: add HTMLDocument

### PR DESCRIPTION
**Pull request by caugner**
[HTMLDocument](https://developer.mozilla.org/en-US/docs/Web/API/HTMLDocument)

Subfeatures (from [W3C Recommendation](https://www.w3.org/TR/2003/REC-DOM-Level-2-HTML-20030109/html.html#ID-26809268)), which are documented on MDN under [Document](https://developer.mozilla.org/en-US/docs/Web/API/Document):
* [title](https://developer.mozilla.org/en-US/docs/Web/API/Document/title)
* [referrer](https://developer.mozilla.org/en-US/docs/Web/API/Document/referrer)
* [domain](https://developer.mozilla.org/en-US/docs/Web/API/Document/domain)
* [URL](https://developer.mozilla.org/en-US/docs/Web/API/Document/URL)
* [body](https://developer.mozilla.org/en-US/docs/Web/API/Document/body)
* [images](https://developer.mozilla.org/en-US/docs/Web/API/Document/images)
* [applets](https://developer.mozilla.org/en-US/docs/Web/API/Document/applets)
* [links](https://developer.mozilla.org/en-US/docs/Web/API/Document/links)
* [forms](https://developer.mozilla.org/en-US/docs/Web/API/Document/forms)
* [anchors](https://developer.mozilla.org/en-US/docs/Web/API/Document/anchors)
* [cookie](https://developer.mozilla.org/en-US/docs/Web/API/Document/cookie)
* `open()`
* `close()`
* `write()`
* `writeln()`
* `getElementsByName()`

If those subfeatures are actually subfeatures of `Document`, we probably don't have to re-document them here, do we?

### CODE REVIEW

**Comment by caugner** [comment]
If those subfeatures are actually subfeatures of `Document`, we probably don't have to re-document them here.

File: api/HTMLDocument.json [hunk 1]
```diff
@@ -0,0 +1,820 @@
...
+            "version_added": null
+          },
+          "safari_ios": {
+            "version_added": null
+          }
+        },
+        "status": {
+          "experimental": false,
+          "standard_track": true,
+          "deprecated": false
```
Comment (a2sheppy) [inline api/HTMLDocument.json]: This interface is deprecated. All of its members are specified to be part of a "partial interface Document" now, so they're actually directly members of the Document interface instead. So "deprecated" should be true throughout the entire file.

File: api/HTMLDocument.json [hunk 2]
```diff
@@ -0,0 +1,820 @@
...
+            "version_added": null
+          },
+          "edge": {
+            "version_added": null
+          },
+          "edge_mobile": {
+            "version_added": null
+          },
+          "firefox": {
+            "version_added": null
```
Comment (a2sheppy) [inline api/HTMLDocument.json]: HTMLDocument is still implemented in Firefox, although some of the members have migrated to Document already. This should be true for both Firefox and Firefox Android throughout this file (unless it's decided to hold that off for another update).

**Comment by Elchi3** [comment]
I don't think it makes sense to add members of a 2003 spec interface which is long time obsolete. Lets not document and not add these members to BCD. I'm reducing the PR to just add the basic support. If browsers still support certain members under this interface we can add them here, but not the ones from an 15 year old outdated spec that no one implements anymore.

**Pull request merged**